Solar-Wrapper/src/TrackablesSolARImpl.cpp

```diff
@@ -0,0 +1,287 @@
+#include "TrackablesSolARImpl.h"
+#include "Trackable.h"
+#include "Helpers.h"
+#include "xpcf/xpcf.h"
+#include "xpcf/core/uuid.h"
+#include "UnitSysConversion.h"
+#include <nlohmann/json.hpp>
+
+namespace xpcf = org::bcom::xpcf;
+namespace org {
+namespace openapitools {
+namespace server {
+namespace implem {
+
+    using namespace org::openapitools::server::model;
+    using namespace SolAR::datastructure;
+    using namespace nlohmann;
+    using namespace Eigen;
+
+    TrackablesSolARImpl::TrackablesSolARImpl(const std::shared_ptr<Pistache::Rest::Router>& rtr, SRef<SolAR::api::storage::IWorldGraphManager> worldStorage)
+        : TrackablesApi(rtr)
+    {
+        m_worldStorage = worldStorage;
+    }
+
+    void TrackablesSolARImpl::add_trackable(const Trackable &trackable, Pistache::Http::ResponseWriter &response)
+    {
+        //convert all the Trackable attributes into StorageTrackable attributes  to create one and store it in the world storage
+
+        //creator uuid
+        xpcf::uuids::uuid creatorId = xpcf::toUUID(trackable.getCreatorUUID());
+
+        //localCRS
+        std::vector<float> vector = trackable.getLocalCRS();
+        float* array = &vector[0];
+        Matrix4f matrix = Map<Matrix4f>(array);
+        Transform3Df localCRS(matrix);
+
+        //unitsystem
+        SolAR::datastructure::UnitSystem unitSystem = resolveUnitSystem(trackable.getUnit());
+
+        //dimension
+        Vector3d dimension = Vector3d(trackable.getTrackableSize().data());
+
+        //parents
+        std::map<xpcf::uuids::uuid, std::pair<SRef<StorageWorldElement>, Transform3Df>> parents{};
+
+        //children
+        std::map<xpcf::uuids::uuid,SRef<StorageWorldElement>> children{};
+
+        //taglist
+        std::multimap<std::string,std::string> keyvalueTagList;
+        for (std::pair<std::string,std::vector<std::string>> tag : trackable.getKeyvalueTags()){
+            for(std::string value : tag.second){
+                keyvalueTagList.insert({tag.first,value});
+            }
+        }
+
+        //trackable type
+        StorageTrackableType type = resolveTrackableType(trackable.getTrackableType());
+
+        //encoding info
+        EncodingInfo encodingInfo(trackable.getTrackableEncodingInformation().getDataFormat(), trackable.getTrackableEncodingInformation().getVersion());
+
+        //payload
+        std::vector<std::byte> payload = TrackablesSolARImpl::toBytes(trackable.getTrackablePayload());
+
+        //adding the newly created StorageTrackable to the worldgraph
+        xpcf::uuids::uuid trackableId;
+        if(m_worldStorage->addTrackable(trackableId, creatorId, localCRS, unitSystem, dimension, parents, children, keyvalueTagList, type, encodingInfo, payload) != FrameworkReturnCode::_SUCCESS)
+        {
+            //if something went wrong
+            response.headers().add<Pistache::Http::Header::ContentType>(MIME(Text, Plain));
```

Comment:
nit: define a method for response ?
```
sendResponse(Code c, string message);
```
As you wish, it just mainly hide one line (header/MIME), so maybe it's not worth it.